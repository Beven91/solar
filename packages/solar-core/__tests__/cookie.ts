import Cookie from '../src/cookie';

describe('Cookie', () => {
  it('parse', () => {
    const str = 'BIDUPSID=18CB9619E5482B2D0E9EC9A37D2CF93B; PSTM=1586315620; BAIDUID=18CB9619E5482B2D1D7FEA5C25651A4D:FG=1; BD_UPN=12314753; COOKIE_SESSION=132245_4_9_7_60_14_0_0_9_5_0_0_131430_0_3_0_1588776067_1587296668_1588776064%7C9%2392079_3_1587291523%7C2; BD_HOME=1; H_PS_PSSID=1429_31125_21107_31589_30826_31463_31228_30823_31164; delPer=0; BD_CK_SAM=1; PSINO=1; BDORZ=B490B5EBF6F3CD402E515D22BCDA1598; BDSVRTM=18';

    const cookies = Cookie.parse(str);

    const runtime = {
      value: '',
    };

    Object.defineProperty(document, 'cookie', {
      set(value) {
        runtime.value = value;
      },
    });


    // 匹配镜像
    expect(cookies).toMatchSnapshot();

    const date = new Date();
    cookies.setCookie('name', 'beven', date, '/');

    // 断言： 底层设置cookie应该正确
    expect(runtime.value).toBe(`name=beven;expires=${date.toUTCString()};path=/`);

    expect(cookies).toMatchSnapshot();

    cookies.setCookie('aaa', null, null);
    expect(cookies).toMatchSnapshot();

    cookies.setCookie(null, null, null);
    expect(cookies).toMatchSnapshot();

    cookies.setCookie('type', '1', new Date(), '/', 'www.solar.com');
    expect(cookies).toMatchSnapshot();

    // 断言：存在 type
    expect(cookies.getCookie('type')).toBe('1');

    // 移除name
    cookies.removeCookie('name');
    expect(cookies).toMatchSnapshot();
  });
});
