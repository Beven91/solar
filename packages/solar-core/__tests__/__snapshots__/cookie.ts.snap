// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cookie parse 1`] = `
Cookie {
  "cookies": Object {
    "BAIDUID": "18CB9619E5482B2D1D7FEA5C25651A4D:FG",
    "BDORZ": "B490B5EBF6F3CD402E515D22BCDA1598",
    "BDSVRTM": "18",
    "BD_CK_SAM": "1",
    "BD_HOME": "1",
    "BD_UPN": "12314753",
    "BIDUPSID": "18CB9619E5482B2D0E9EC9A37D2CF93B",
    "COOKIE_SESSION": "132245_4_9_7_60_14_0_0_9_5_0_0_131430_0_3_0_1588776067_1587296668_1588776064%7C9%2392079_3_1587291523%7C2",
    "H_PS_PSSID": "1429_31125_21107_31589_30826_31463_31228_30823_31164",
    "PSINO": "1",
    "PSTM": "1586315620",
    "delPer": "0",
  },
}
`;

exports[`Cookie parse 2`] = `
Cookie {
  "cookies": Object {
    "BAIDUID": "18CB9619E5482B2D1D7FEA5C25651A4D:FG",
    "BDORZ": "B490B5EBF6F3CD402E515D22BCDA1598",
    "BDSVRTM": "18",
    "BD_CK_SAM": "1",
    "BD_HOME": "1",
    "BD_UPN": "12314753",
    "BIDUPSID": "18CB9619E5482B2D0E9EC9A37D2CF93B",
    "COOKIE_SESSION": "132245_4_9_7_60_14_0_0_9_5_0_0_131430_0_3_0_1588776067_1587296668_1588776064%7C9%2392079_3_1587291523%7C2",
    "H_PS_PSSID": "1429_31125_21107_31589_30826_31463_31228_30823_31164",
    "PSINO": "1",
    "PSTM": "1586315620",
    "delPer": "0",
    "name": "beven",
  },
}
`;

exports[`Cookie parse 3`] = `
Cookie {
  "cookies": Object {
    "BAIDUID": "18CB9619E5482B2D1D7FEA5C25651A4D:FG",
    "BDORZ": "B490B5EBF6F3CD402E515D22BCDA1598",
    "BDSVRTM": "18",
    "BD_CK_SAM": "1",
    "BD_HOME": "1",
    "BD_UPN": "12314753",
    "BIDUPSID": "18CB9619E5482B2D0E9EC9A37D2CF93B",
    "COOKIE_SESSION": "132245_4_9_7_60_14_0_0_9_5_0_0_131430_0_3_0_1588776067_1587296668_1588776064%7C9%2392079_3_1587291523%7C2",
    "H_PS_PSSID": "1429_31125_21107_31589_30826_31463_31228_30823_31164",
    "PSINO": "1",
    "PSTM": "1586315620",
    "aaa": null,
    "delPer": "0",
    "name": "beven",
  },
}
`;

exports[`Cookie parse 4`] = `
Cookie {
  "cookies": Object {
    "BAIDUID": "18CB9619E5482B2D1D7FEA5C25651A4D:FG",
    "BDORZ": "B490B5EBF6F3CD402E515D22BCDA1598",
    "BDSVRTM": "18",
    "BD_CK_SAM": "1",
    "BD_HOME": "1",
    "BD_UPN": "12314753",
    "BIDUPSID": "18CB9619E5482B2D0E9EC9A37D2CF93B",
    "COOKIE_SESSION": "132245_4_9_7_60_14_0_0_9_5_0_0_131430_0_3_0_1588776067_1587296668_1588776064%7C9%2392079_3_1587291523%7C2",
    "H_PS_PSSID": "1429_31125_21107_31589_30826_31463_31228_30823_31164",
    "PSINO": "1",
    "PSTM": "1586315620",
    "aaa": null,
    "delPer": "0",
    "name": "beven",
  },
}
`;

exports[`Cookie parse 5`] = `
Cookie {
  "cookies": Object {
    "BAIDUID": "18CB9619E5482B2D1D7FEA5C25651A4D:FG",
    "BDORZ": "B490B5EBF6F3CD402E515D22BCDA1598",
    "BDSVRTM": "18",
    "BD_CK_SAM": "1",
    "BD_HOME": "1",
    "BD_UPN": "12314753",
    "BIDUPSID": "18CB9619E5482B2D0E9EC9A37D2CF93B",
    "COOKIE_SESSION": "132245_4_9_7_60_14_0_0_9_5_0_0_131430_0_3_0_1588776067_1587296668_1588776064%7C9%2392079_3_1587291523%7C2",
    "H_PS_PSSID": "1429_31125_21107_31589_30826_31463_31228_30823_31164",
    "PSINO": "1",
    "PSTM": "1586315620",
    "aaa": null,
    "delPer": "0",
    "name": "beven",
    "type": "1",
  },
}
`;

exports[`Cookie parse 6`] = `
Cookie {
  "cookies": Object {
    "BAIDUID": "18CB9619E5482B2D1D7FEA5C25651A4D:FG",
    "BDORZ": "B490B5EBF6F3CD402E515D22BCDA1598",
    "BDSVRTM": "18",
    "BD_CK_SAM": "1",
    "BD_HOME": "1",
    "BD_UPN": "12314753",
    "BIDUPSID": "18CB9619E5482B2D0E9EC9A37D2CF93B",
    "COOKIE_SESSION": "132245_4_9_7_60_14_0_0_9_5_0_0_131430_0_3_0_1588776067_1587296668_1588776064%7C9%2392079_3_1587291523%7C2",
    "H_PS_PSSID": "1429_31125_21107_31589_30826_31463_31228_30823_31164",
    "PSINO": "1",
    "PSTM": "1586315620",
    "aaa": null,
    "delPer": "0",
    "name": "",
    "type": "1",
  },
}
`;
